### PR TITLE
Source Facebook Marketing: automatically reduce batch size

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve https://github.com/airbytehq/oncall/issues/2765

## How

Automatically reduce batch size `/ 2` if Too much data was requested in batch.

set maximum for `max_batch_size` = 50 (Facebook limitation)

## Recommended reading order
1. `base_streams.py`
2. `spec.json`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_streams.py [hunk 1]
```diff
@@ -78,9 +78,12 @@ def failure(response: FacebookResponse, request: Optional[FacebookRequest] = Non
             # although it is Optional in the signature for compatibility, we need it always
             assert request, "Missing a request object"
             resp_body = response.json()
-            if not isinstance(resp_body, dict) or resp_body.get("error", {}).get("code") != FACEBOOK_BATCH_ERROR_CODE:
+            if not isinstance(resp_body, dict) or (resp_body.get("error", {}).get("code") != FACEBOOK_BATCH_ERROR_CODE and resp_body.get("error", {}).get("message") != "Please reduce the amount of data you're asking for, then retry your request"):
```
Comment (girarda) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_streams.py:81]: we're doing a similar check twice and the flow is difficult to follow

can we rewrite as 
1. If resp_body is not a dict: raise RuntimeError
2. elif error message == please reduce ...: update the batch size
3. elif error code != FACEBOOK_BATCH_ERROR: raise RuntimeError

two branches will lead to the same RuntimeError, but the conditions will be easier to read

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_streams.py [hunk 2]
```diff
@@ -78,9 +78,12 @@ def failure(response: FacebookResponse, request: Optional[FacebookRequest] = Non
...
                 # response body is not a json object or the error code is different
                 raise RuntimeError(f"Batch request failed with response: {resp_body}")
+            if resp_body.get("error", {}).get("message") == "Please reduce the amount of data you're asking for, then retry your request":
```
Comment (girarda) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_streams.py:84]: ok to do in a separate PR since this is fixing a P0, but it would be useful to unit test this method

**Review by lazebnyi — APPROVED** [review]
